__tests__/classes/noop-logger-factory.ts

```diff
@@ -0,0 +1,32 @@
+import {Logger} from '../../src/classes/loggers/logger';
+import {LoggerFactory} from '../../src/classes/loggers/logger-factory';
+
+/**
+ * @description Creates Logger instances that won't generate any logs.
+ * Useful for getting rid of verbose output while testing.
+ */
+export class NoopLoggerFactory extends LoggerFactory {
```

Comment:
What do you do in the case where you wish to verify that the logs are ok?

Reply:
@C0ZEN I guess it depends what you mean with ok. The `Logger` class(es) test suites should catch most logger related bugs. If you want to test something like `it('method f() should print "f() called"', () => ...);` you could inject your own `loggerFactoryMock` with the necessary spies into your SUT. What do you think? What I'm really trying to avoid is having every test case (even the ones not related to logging at all!) spamming the console.

Reply:
@joeveiga well I am working on this one https://github.com/actions/stale/pull/358 and as you can see it's mostly logs enhancement.
Having the tests showing the logs as if it's the normal stale action is very useful.
I am not against hiding the logs by default in the tests but IMO we should then have a simple feature like a boolean to switch between between on/off.

Reply:
Eventually, you could still provide a mocked class of the Logger calling the real code with `super` but adding an extra feature with a boolean which is `isEnabled` so that we can toggle on/off the logs.
This boolean could be set on the jest env files maybe?

Reply:
@C0ZEN I see. It looks neat, btw! So, if you have a test suite (or test case) that would benefit from outputting the logs, couldn't you inject the regular `LoggerFactory` for that particular suite/case, instead of the `NoopLoggerFactory`? My only concern with printing to the console by default for the tests is 1) my jest output gets lost in all the noise, and 2) writing to the console is technically (kinda) like writing to a file or a db, that's not a unit test at that point.
IMO a feature to switch logging on/off could be useful if that's a config option we're going to expose to the users, otherwise probably not so much.

Reply:
I reply because I dunno if you had the time to read my previous message but:
- Yes, having the logs hell in the test is not good so we can by default get rid of them (all)
- But we need sometimes to enabled them all once again so
- If we replace the Logger class with a NoopLogger in the jest setup https://jestjs.io/docs/en/configuration#globalsetup-string which has a property which can toggle the log it would be very nice IMO

Reply:
In any case, if you prefer a flag as opposed to the factory we can do that too. Maybe something like:

```ts
export class X {
  constructor(logger: Logger) {}
  // ...
}

// logger
export class Logger {
  constructor(private readonly output: boolean = true) {}
}

// spec file
const x = new X(new Logger(false));
```

Reply:
Ah, got it! Yeah that sounds good. Sorry I missed the last message 😄

Reply:
Maybe provide the PR like this and I will take a look later about what can we do to simplify?